packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -1501,6 +1526,18 @@ export class ProjectService extends BaseService {
             throw new ForbiddenError();
         }
 
+        if (
+            metricQuery.customDimensions?.some(isCustomSqlDimension) &&
```

Comment:
concerned about how many times we have to write this down

Reply:
me too. Project service needs to be refactored and query logic extracted from it so it can be used across multiple services.

Reply:
Do you think there are places here that might stop viewers from just "viewing" a chart with custom dimensions? It should only stop people if they're able to directly submit a metricQuery themselves

Reply:
Yes. Underlying data, explore from here, drill by will throw errors as they generate new metric queries. 
We will have to work on a better UX in such cases. example: remove custom SQL dimensions if they have no permission instead of throwing an error.

Reply:
Are the metric queries generated client side for these interactions?

Reply:
yes, client side. We would have to change the API to build them on the backend to make the underlying data and drill by work. Explore from here might be harder unless we create a "created from X" and then make sure they can only use the custom dimensions from that saved chart and not new ones.

Reply:
Yeah good point. Looks like underlying data should be able to work easily. But drill by (which is basically explore from here) and explore from here, are not going to work without much more complex permissions.